roles/testnode/files/libexec/smart.sh

```diff
@@ -0,0 +1,284 @@
+#!/bin/bash
+# Description:	Bash script to check drive health using pending, uncorrectable,
+# and reallocated sector count
+#
+# Nagios return codes: 0 = OK; 1 = WARNING; 2 = CRITICAL; 3 = UNKNOWN
+# SMART Attribute Codes:
+#   5 = Reallocated
+#   187 = Reported Uncorrect
+#   197 = Pending
+#   198 = Uncorrectable Sector Count
+#
+# TO-DO: Add support for dynamic SMART attribute lookup.  For example,
+#        187 is reported for Seagate HDD and all SSDs but not Hitachi HDDs.
+#
+# See https://en.wikipedia.org/wiki/S.M.A.R.T.#ATA_S.M.A.R.T._attributes
+
+### Pre-flight checks ###
+# Set raid var then check for cli64 command and bail if missing
+if lspci | grep -qi areca
+then
+  raid=true
+else
+  raid=false
+fi
+
+if [ "$raid" = true ] && ! [ -x "$(command -v cli64)" ]
+then
+  echo "ERROR - cli64 command not found or is not executable"
+  exit 3
+fi
+
+# Check for smartmontools and bail if missing
+if ! [ -x "$(command -v smartctl)" ]
+then
+  echo "ERROR - smartctl is not installed or is not executable"
+  echo "yum/apt-get install smartmontools"
+  exit 3
+fi
+
+### Define global variables ###
+# total number of drives (or RAID slots) discovered
+numdrives=0
+# Number of failed, failing, and/or missing drives
+failingdrives=0
+# Fallback message for UNKNOWN return code output
+unknownmsg="Unknown error"
+# Return code for nagios (Default to UNKNOWN)
+rc=3
+# Array of messages indicating drive health.  Output after nagios status.
+declare -a messages
+
+### Functions ###
+main ()
+{
+  if [ "$raid" = true ]
+  then
+    areca_smart
+    areca_failed
+  elif [ "$raid" = false ]
+  then
+    normal_smart
+  else
+    echo "ERROR - Could not determine if RAID present"
+    exit 3
+  fi
+}
+
+# Gather smart data for drives behind Areca RAID controller
+areca_smart ()
+{
+  # Store output of cli64 to reduce repeated executions
+  cli64out=$(sudo cli64 disk info | grep -E "Slot#[[:digit:]]")
+  numdrives=$(echo "$cli64out" | wc -l)
```

Comment:
Really just musing that it could be numdrives=$(wc -l <<<"$cli64out")....although maybe it couldn't because maybe that would join lines.  anyway, ignore it.